unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarMeshCombiner/AvatarMeshCombinerHelper.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Assertions;
+using Object = UnityEngine.Object;
+
+namespace DCL
+{
+    /// <summary>
+    /// AvatarMeshCombinerHelper uses the AvatarMeshCombiner utility class to combine many skinned renderers
+    /// into a single one.
+    ///
+    /// This class will recycle the same gameObject and renderer each time it is called,
+    /// and binds the AvatarMeshCombiner output to a proper well configured SkinnedMeshRenderer. 
+    /// </summary>
+    public class AvatarMeshCombinerHelper : IDisposable
+    {
+        private static bool VERBOSE = false;
+        private static ILogger logger = new Logger(Debug.unityLogger.logHandler) { filterLogType = VERBOSE ? LogType.Log : LogType.Warning };
+
+        public GameObject container { get; private set; }
+        public SkinnedMeshRenderer renderer { get; private set; }
+
+        public AvatarMeshCombinerHelper (GameObject container = null)
+        {
+            this.container = container;
+        }
+
+        /// <summary>
+        /// Combine will use AvatarMeshCombiner to generate a combined avatar mesh.
+        /// After the avatar is combined, it will generate a new GameObject with a SkinnedMeshRenderer that contains
+        /// the generated mesh. This GameObject and Renderer will be preserved over any number of Combine() calls.
+        /// 
+        /// For more details on the combining check out AvatarMeshCombiner.CombineSkinnedMeshes.
+        /// </summary>
+        /// <param name="bonesContainer">A SkinnedMeshRenderer that must contain the bones and bindposes that will be used by the combined avatar.</param>
+        /// <param name="renderersToCombine">A list of avatar parts to be combined</param>
+        /// <param name="materialAsset">A material asset that will serve as the base of the combine result. A new materialAsset will be created for each combined sub-mesh.</param>
+        /// <returns>true if succeeded, false if not</returns>
+        public bool Combine(SkinnedMeshRenderer bonesContainer, SkinnedMeshRenderer[] renderersToCombine, Material materialAsset)
+        {
+            Assert.IsTrue(bonesContainer != null, "bonesContainer should never be null!");
```

Comment:
Love the use of assertions. We could also use `Assert.NotNull`.

Reply:
The thing is that `Assert.NotNull` doesn't use the overriden equality operator of `UnityEngine.Object` objects, and thus can give false negatives/positives in some weird situations. That forces me to use the `IsTrue` assertion every time.